### ISSUE TITLE
Soldeer compatibility / remappings issue

### ISSUE THREAD

**Issue by PatrickAlphaC** [Cyfrin/aderyn#712]
**Is your feature request related to a problem? Please describe.**
Right now, if you run aderyn on a repository that uses soldeer as it's package manager, it will fail.

**Describe the solution you'd like**
For aderyn to work with soldeer. 

# Walkthrough

```
git clone https://github.com/Cyfrin/2024-09-president-elector
soldeer install 
forge build
aderyn .
```

You will get an error because it doesn't recognize dependencies in a different folder.

**Comment by TilakMaddy** [comment]
## Temporary solution:


If you remove

```
 [dependencies]
"@openzeppelin-contracts" = "5.0.2"
```

from foundry.toml and run `aderyn`, it will work.

**Comment by TilakMaddy** [comment]
@alexroan looks like we need to update our `foundry-config` fork again (not foundry-compilers-backend) because its unable to parse the dependency above and it errors out. Maybe the new updated version allows this style

**Comment by TilakMaddy** [comment]
@PatrickAlphaC we don't assume  _any_ framework or _package manager_  in the user's environment :) This was a config parse error from what it looks like to me ...

**Comment by TilakMaddy** [comment]
# Aderyn Analysis Report

This report was generated by [Aderyn](https://github.com/Cyfrin/aderyn), a static analysis tool built by [Cyfrin](https://cyfrin.io), a blockchain security company. This report is not a substitute for manual audit or security review. It should not be relied upon for any purpose other than to assist in the identification of potential security vulnerabilities.
# Table of Contents

- [Summary](#summary)
  - [Files Summary](#files-summary)
  - [Files Details](#files-details)
  - [Issue Summary](#issue-summary)
- [High Issues](#high-issues)
  - [H-1: Storage Array Edited with Memory](#h-1-storage-array-edited-with-memory)
- [Low Issues](#low-issues)
  - [L-1: Loop condition contains `state_variable.length` that could be cached outside.](#l-1-loop-condition-contains-statevariablelength-that-could-be-cached-outside)
  - [L-2: Costly operations inside loops.](#l-2-costly-operations-inside-loops)


# Summary

## Files Summary

| Key | Value |
| --- | --- |
| .sol Files | 1 |
| Total nSLOC | 154 |


## Files Details

| Filepath | nSLOC |
| --- | --- |
| src/RankedChoice.sol | 154 |
| **Total** | **154** |


## Issue Summary

| Category | No. of Issues |
| --- | --- |
| High | 1 |
| Low | 2 |


# High Issues

## H-1: Storage Array Edited with Memory

Storage reference is passed to a function with a memory parameter. This will not update the storage variable as expected. Consider using storage parameters instead.

<details><summary>3 Found Instances</summary>


- Found in src/RankedChoice.sol [Line: 73](src/RankedChoice.sol#L73)

	```solidity
	                if (!_isInArray(s_candidateList, orderedCandidates[j])) {
	```

- Found in src/RankedChoice.sol [Line: 79](src/RankedChoice.sol#L79)

	```solidity
	        address[] memory winnerList = _selectPresidentRecursive(
	```

- Found in src/RankedChoice.sol [Line: 167](src/RankedChoice.sol#L167)

	```solidity
	        if (!_isInArray(VOTERS, voter)) {
	```

</details>



# Low Issues

## L-1: Loop condition contains `state_variable.length` that could be cached outside.

Cache the lengths of storage arrays if they are used and not modified in for loops.

<details><summary>2 Found Instances</summary>


- Found in src/RankedChoice.sol [Line: 68](src/RankedChoice.sol#L68)

	```solidity
	        for (uint256 i = 0; i < VOTERS.length; i++) {
	```

- Found in src/RankedChoice.sol [Line: 107](src/RankedChoice.sol#L107)

	```solidity
	        for (uint256 i = 0; i < VOTERS.length; i++) {
	```

</details>



## L-2: Costly operations inside loops.

Invoking `SSTORE`operations in loops may lead to Out-of-gas errors. Use a local variable to hold the loop computation result.

<details><summary>4 Found Instances</summary>


- Found in src/RankedChoice.sol [Line: 68](src/RankedChoice.sol#L68)

	```solidity
	        for (uint256 i = 0; i < VOTERS.length; i++) {
	```

- Found in src/RankedChoice.sol [Line: 72](src/RankedChoice.sol#L72)

	```solidity
	            for (uint256 j = 0; j < orderedCandidates.length; j++) {
	```

- Found in src/RankedChoice.sol [Line: 107](src/RankedChoice.sol#L107)

	```solidity
	        for (uint256 i = 0; i < VOTERS.length; i++) {
	```

- Found in src/RankedChoice.sol [Line: 108](src/RankedChoice.sol#L108)

	```solidity
	            for (
	```

</details>

**Comment by PatrickAlphaC** [comment]
Nice! Ok so... is this going to stay an issue then?

**Comment by TilakMaddy** [comment]
Yes sir! Until we Paras the config properly its going to remain an open issue

**Comment by TilakMaddy** [comment]
GM @PatrickAlphaC ! Finally solved in 0.5.4 .I've verified it 🙌

**Comment by PatrickAlphaC** [comment]
lfg!
